Refresh QUICKSTART.rst and building.rst

bradcray opened the pull request:
Motivated by user issue #18150, I took an editing pass over QUICKSTART.rst and building.rst to refresh them for the upcoming Chapel 1.25 release.  Specifically, my high-level goals were to:

* keep QUICKSTART as lean and mean as possible while having it refer users to other places to avoid common traps
* make sure that things were still as accurate as possible
* promote the `make install` content a bit higher and address the challenge of using Chapel across multiple shells more directly
* generally wordsmithed and smoothed things over

In more detail, In `building.rst`, I:

* removed the reference back to QUICKSTART.rst in this file, promoting chplenv.rst in its place, because I think quickstart should be all about "quick and dirty" with building.rst and chplenv.rst being more about "the complete story in plenty of detail"
* mentioned `printchplenv` in the opening paragraph for someone arriving in this file without going through `chplenv.rst` to understand what we mean by the Chapel environment
* moved the comment about interpreting the output of `make -v` into actual text in the document because if someone needs this level of hand-holding, they may not know what a bash comment is and may type it in
* fixed the updated path to `chpl`
* fixed the prefix of the path to the libraries (while leaving the details vague)
* moved the part about each `make` command only building the current configuration further down in the document
* made the part about "now you can compile" allude to the next few sections so that people wouldn't stop reading prematurely (while still knowing that they had a working compile)
* added a detailed section about using the built Chapel in other shells / terminals, because this is increasingly FAQ'd as we get more beginning users
* fleshed out the section of `./configure` + `make install` to be a bit more precise about how to use it.  I can never remember, and I feel as though those commands currently require more than an ideal amount of expertise.  I think we could go even further with hand-holding in these instructions, but I'm not quite clear I could do it justice (not really understanding the full behavior...), and hopefully this is a step in the right direction.
* took out the paragraph about supported platforms because... what is that doing here?  (and who wants to maintain it here)
* added a section about switching between multiple configurations / supporting them in a single place
* removed a big section about supporting a new compiler or platform by simply dropping in a Makefile.<whatever> because I don't think the world is that simple anymore (at least on the compiler side of things)
* tweaked some details and formatting and wordings in the Makefile-related sections

In QUICKSTART.rst, I:

* started the document off with a bit more of "What the goal here is", referring people to the "official / full-fledged" instructions if they'd prefer to do that and trying to make the quickstart vs. full-featured branch clearer earlier.  Overall, my hope is that most people will head into the quickstart instructions due to lack of patience, but that those patient enough to read these paragraphs may self select into the chplenv + building docs or perhaps jump directly to the preferred configuration if they don't want to waste time with a stripped-down version / are willing to deal with more detail.  This also allowed me to make subsequent bullets shorter and sweeter (where my guess is that 80% of readers will just jump to step 0 and start typing stuff... I know I would).
* changed from "expand" to "unpack" when talking about dealing with the .tar.gz
* generally switched from hyperlinks that refer to filenames and section names to those that spell out the section (with the assumption that it looks better in web form and that that's where most people see it)
* removed references to `make check` since it essentially builds and runs a program which is what the instructions are just about to suggest you do
* switched hello.chpl to hello3-datapar.chpl just because it's more interesting; may produce more of a "wow!" factor when run; may lead people to look at what hellos 1-3+ are, etc.
* removed the -o flag from the `chpl` command lines since we don't generate `a.out` anymore, so it seems redundant
* tried to make the section about LLVM in the preferred configuration a bit clearer by breaking the options out into bullets
* forked the part about using your build in other shells/terminals into its own section (for visibility given that it's FAQ'd) but then had it refer to the relevant section in building.rst to avoid duplication / keep this treatment short and sweet
* fixed the link about performance tips to point to the more specific page
* added a note for reassurance that we support the setchplenv scripts for the preferred mode for non-bash shells as well

Resolves #18150.

mstrout commented on doc/rst/usingchapel/QUICKSTART.rst:
```diff
@@ -117,40 +116,42 @@ you will need to rebuild Chapel from source in a different configuration.
+different shell or terminal, you will need to either re-``source`` the
+``setchplenv.bash`` script that you used when building Chapel or set
+up your environment to support additional shells automatically.  See
+:ref:`using-chapel-in-other-sessions` for details.
```
Is another layer of indirection needed here?  Can't they just say `make install`?

bradcray replied:
I think you also have to run `./configure` before making anything in order for `make install` to work, which requires specifying one of the install path forms if you don't want the default of `/usr/local` (which I'd say isn't an appropriate default for a QUICKSTART user who's likely to get a permissions error and be confused by it).  And I don't really want the QUICKSTART to have to teach the user about all of that—at least this week and until/unless we have more / better testing of it and confidence in it.

Note that `./configure` and `make install` can be run after all of the instructions above with very little time spent or downsides (specifically, it won't generally re-do the whole `make` from scratch, so I don't think much is lost by mentioning it late / doing it after the fact).

mppf replied:
As it stands now, I worry that it will lead a user to think that installing is not possible. (Because it implies that the *only* way to have it work in different terminals is by sourcing something possibly from bashrc / changing environment variables).

In particular, I think part of the problem is that I interpret this section

```
To use Chapel from a
different shell or terminal, you will need to either re-``source`` the
``setchplenv.bash`` script that you used when building Chapel or set
up your environment to support additional shells automatically.  See
:ref:`using-chapel-in-other-sessions` for details.
```

as not including `make install` as a possibility. Perhaps you were imagining that `make install` is a way to "set up your environment"?

Instead, I think we should fix it by relying more on the linked text - replacing the above quoted section with this:

```
One easy way to use Chapel from a
different shell or terminal is to re-``source`` the
``setchplenv.bash`` script that you used when building Chapel.
However since that can be annoying to do, there are several
other strategies available to you. Please see :ref:`using-chapel-in-other-sessions`
to learn about them.
```

mstrout commented on doc/rst/usingchapel/QUICKSTART.rst:
```diff
@@ -117,40 +116,42 @@ you will need to rebuild Chapel from source in a different configuration.
-:ref:`multilocale.rst <readme-multilocale>`.
+All of the instructions above describe how to run Chapel programs in a
+single-locale (shared-memory) mode. To run using multiple locales
+(distributed memory), please refer to :ref:`readme-multilocale`.
```
Hmmm, so running in single locale mode is the "preferred, full-feature mode"?  Maybe above when we say preferred mode, we could say the preferred mode on a single node?  It would be a bit annoying for an HPC programmer to get to this point and realize they need to rebuild Chapel yet again to get what they want.

bradcray replied:
Good point.  I've inserted a bullet before the "Repeat steps 2–5 above" bullet that says:

> * If you are interested in building Chapel to support multiple locales
>   (distributed memory execution), refer to :ref:`readme-multilocale`
>   for other settings to	enable that.

mstrout commented on doc/rst/usingchapel/QUICKSTART.rst:
```diff
@@ -117,40 +116,42 @@ you will need to rebuild Chapel from source in a different configuration.
+--------------------
+
+If you plan to do performance studies of Chapel programs, be sure to
+use the full-featured version above and see
```
Except for when you want performance on a machine with multiple nodes?  Also, we should consider using the term node when we use the term locale, since people just getting started with Chapel won't know the correspondence there.

bradcray replied:
> Except for when you want performance on a machine with multiple nodes?

I think of this comment as being orthogonal to the multiple node question (where I think of its main purpose as being "don't do performance studies with the quickstart settings" and from there, you're welcome to do either single- or multiple-node performance studies).  (?)

mstrout replied:
Previously the "full-featured version" was implicitly referring to single node.  If you made that explicit, then this is fine.

mstrout commented on doc/rst/usingchapel/QUICKSTART.rst:
```diff
@@ -167,6 +168,9 @@ the fish shell (fish)                ``. util/quickstart/setchplenv.fish``
 the Bourne shell (sh)                ``. util/quickstart/setchplenv.sh``
 ==================================== ==========================================
 
+For the full-featured builds using these shells, simply remove
```
For the full-featured, single-locale builds..."

bradcray replied:
I'd rather not have to add a single-locale caveat each time we mention any of these settings, so adjusted the first paragraph to state the qualification early in order to avoid continuing to return to it (other than the bullet and section that explicitly point to the multi-locale settings documentation, which is not trivial / quickstart by nature):

> These instructions are designed to help users get started with a
> single-locale (shared-memory) implementation of Chapel from the source
> distribution of the release as quickly as possible.  If you are
> interested in taking the time to understand Chapel's configuration
> options, build process, and installation more thoroughly, please refer
> to :ref:`readme-chplenv` and :ref:`readme-building` instead.

(I'm also not sure that a multi-locale build is _always_ our preferred configuration.
e.g., if you're only running on shared-memory, we'd prefer you to use a single-locale
build to avoid the overheads that `--no-local` introduces).

mstrout commented on doc/rst/usingchapel/building.rst:
```diff
@@ -5,19 +5,22 @@ Building Chapel
 To build the Chapel compiler, set up your environment as described in
-:ref:`chapelhome-quickstart` (or :ref:`readme-chplenv` for more
-settings), cd to ``$CHPL_HOME``, and run GNU make.
+:ref:`readme-chplenv`, cd to ``$CHPL_HOME`` (the root directory of
```
Can this be more self contained by saying how to set up the environment for the preferred, single-locale Chapel compiler and runtime and referring them to `readme-chplenv` if they want something other than that?

bradcray replied:
The first sections of `readme-chplenv` already talk about the preferred environment pretty directly and succinctly (which is not to say it couldn't be improved or streamlined further, but that's a separate task).  So if we were to say more here rather than redirecting, I fear we get into the case of every file trying to say everything redundantly rather than each playing its position (where I view this as being designed to be the second file you'd read after that one).

One of the ironic things about the preferred, single-locale Chapel build (e.g., `setchplenv.bash`) is that it doesn't actually set anything interesting because the `CHPL_*` variables are designed to default to reasonable values.  Specifically, it sets:
* CHPL_HOME (which might even no longer be necessary... we've removed most dependencies on it in recent years)
* PATH to include the `bin/` directory
* MANPATH to include the `man/` directory
These are called out in the `Recommended Settings` portion of `building.rst` directly and indirectly via the mention of the `setchplenv.*` scripts in the second paragraph.

mstrout replied:
Ok.

mstrout commented on doc/rst/usingchapel/building.rst:
```diff
@@ -62,132 +56,149 @@ display its help message using:
+the current set of ``CHPL_`` environment variables defined by—and
+inferred for—your environment.  To build support for additional
+configurations, you will need to modify your ``CHPL_`` environment
+variables and re-make.
```
So if I built different configurations in different locations, how would I switch between them?

bradcray replied:
Good point.  I added a second paragraph to address that:

> Once you have built multiple configurations, you can switch between
> them either by setting your ``CHPL_`` environment variables to the
> same values as when the configuration was built, or using the ``chpl``
> compiler's command-line options that correspond to the environment
> variables (see the 'Compiler Configuration Options' section of the
> :ref:`man-chpl` man page for details).

mstrout reviewed (commented):
Overall this is a nice edit to the quickstart and installation documentation.  There should be a bit more discussion on when to mention the instances across terminals piece in quickstart, but the details about it in building.rst are great.

Process question: I am marking my review as a "Comment", but do we typically have people mark a review as "Request changes" or "Approve" when there is substantive feedback?

bradcray commented on doc/rst/usingchapel/QUICKSTART.rst:
```diff
@@ -3,75 +3,70 @@
-enable more features, such as distributed memory execution.
+These instructions first have you build a minimal, stripped-down
+version of Chapel to reduce build times and the potential for
+confusion.  Once you are interested in a full-featured version of
```
I actually am interested in reducing user confusion (and think things like choosing whether to opt into or out of LLVM have the potential to cause confusion), but I agree that it may come across as judgmental / a chip to knock off of someone's shoulder.  For brevity how about just "to reduce build times and the potential for portability issues." ?

mstrout replied:
How about take off "and the potential for portability issues."?  I prefer we not suggest to people that Chapel might not be portable, when the issue really is concern about other packages.

bradcray replied:
That's a good point.  Maybe "third-party portability issues" to try and get the best of both worlds?

mstrout replied:
That would work.

bradcray commented on doc/rst/usingchapel/QUICKSTART.rst:
```diff
@@ -3,75 +3,70 @@
+      chpl examples/hello3-datapar.chpl
+
 
 4) Run the resulting executable:
```
To keep this within a single 80-column line, how about:

```
3) Compile an example program, which uses a ``forall`` loop to print messages:
...
4) Run the resulting executable which will print 100 messages in parallel:
```

mstrout replied:
Sure, but you have to replace "which" with "that" if you take out the comma.

bradcray replied:
To be clear, do you prefer the "that will" form, or are you passively suggesting I should add the missing comma?  (taking it out was not intentional—there wasn't one before to take out).

mstrout replied:
Thank you for putting the comma in.

bradcray commented on doc/rst/usingchapel/QUICKSTART.rst:
```diff
@@ -3,75 +3,70 @@
+      ./hello3-datapar
+
 
 5) Experiment with Chapel in this Quickstart mode to your heart's
```
> Or could there at least be a forward reference to "Using Chapel in a Different Shell / Session" below?

Sounds good, see my next commit.

bradcray commented on doc/rst/usingchapel/QUICKSTART.rst:
```diff
@@ -81,33 +76,37 @@ enable more features, such as distributed memory execution.
+rebuild Chapel from source in a different configuration:
 
 *  Open up a new shell to avoid inheriting the previous environment
    settings.
```
I don't think that `util/setchplenv.bash` unsets anything explicitly, but only sets things.  Whereas `util/quickstart/setchplenv.bash` does set things that would need to be unset to get to the preferred configuration. 
 And I think this is appropriate—I wouldn't want to have to maintain the scripts over time to unset everything that someone may have set previously.  So tI thinkfiring up a new shell is the simplest way for a new/naive user to wipe the slate clean.

mstrout replied:
got it.  Ok sounds good.

bradcray commented on doc/rst/usingchapel/QUICKSTART.rst:
```diff
@@ -81,33 +76,37 @@ enable more features, such as distributed memory execution.
+  1.25, LLVM is the default back-end, which needs to be available for
+  full functionality.  There are a few ways to use LLVM:
+
+  - ensure that you have a version of LLVM 11 installed on your system
```
It has to be LLVM 11, I believe.  The main thing I worry about in saying this here is that it's another place in the documentation that a version number will be maintained.  In other places, we say things like "a supported version of LLVM", but it felt a bit unfair/vague to do this in a quickstart that is generally intended to stand alone.

bradcray commented on doc/rst/usingchapel/QUICKSTART.rst:
```diff
@@ -81,33 +76,37 @@ enable more features, such as distributed memory execution.
+    find it if it's in your path)
+
+  - set ``CHPL_LLVM=bundled`` to have Chapel build and use the bundled
+    version of LLVM (note that this can take a *long* time)
```
I used:

>  (note that building the bundled version of LLVM can take a *long* time)

because it's not really the building of the Chapel compiler itself that takes a particularly long time, but the LLVM stage itself.

bradcray commented:
> I am marking my review as a "Comment", but do we typically have people mark a review as "Request changes" or "Approve" when there is substantive feedback?

I don't think we've established a standard practice (I only even learned where the "Approve" button was recently), but I don't normally select "Approve" or indicate approval unless either (a) it doesn't matter to me whether or not they respond to my comments or (b) it's above the bar and I don't feel the need to review it further.  In this case, I wouldn't have hit "Approve" if I'd been in your shoes.

bradcray commented:
Hi Michelle — I think I've addressed all your comments (and caught some other issues) if you have time to re-read.  If not, let me know and I'll get a second review from someone else (which I might do anyway given the importance of the files).

bradcray commented on doc/rst/usingchapel/chplenv.rst:
```diff
@@ -687,6 +686,9 @@ CHPL_LLVM
        none           do not support llvm/clang-related features
+       unset          indicates that no reasonable default has been
+                      inferred, requiring the user to intentionally select
+                      another option
```
@mppf / @daviditen (upon returning): while working on CHANGES.md today, I noticed that `unset` was not described in this table which seemed asymmetrical to other cases to me (I tend to think of the tables as listing all of the options; I realize this case is a little asymmetrical in that we'd never expect anyone to purposefully set it).  But I wanted to make sure that this wasn't left off on purpose for reasons I wasn't picking up on.  Thanks.

mppf replied:
FWIW, I think it's fine and reasonable to add it to the table

mstrout commented on doc/rst/usingchapel/building.rst:
```diff
@@ -62,132 +56,156 @@ display its help message using:
+
+.. code-block:: bash
+
+  ./configure  # use ``./configure --help`` for specific options
```
Take the backquotes out of the ./configure --help in the comment, maybe just use regular quotes?  It looks strange when viewing the file from https://github.com/chapel-lang/chapel/blob/f1395f93db429732b760b7d0ad7f9127d42f27aa/doc/rst/usingchapel/building.rst.

bradcray replied:
Good catch!

mstrout commented:
I just re-read the whole quickstart.rst and building.rst files.  I found one minor formatting suggestion in the building.rst file and marked it.  Other than that it looks great, and I think this PR is ready to merge.

bradcray commented:
Thanks Michelle!

e-kayrakli commented on doc/rst/usingchapel/QUICKSTART.rst:
```diff
@@ -3,161 +3,177 @@
+These instructions are designed to help users get started with a
+single-locale (shared-memory) implementation of Chapel from the source
+distribution of the release as quickly as possible.  If you are
+interested in taking the time to understand Chapel's configuration
```
"interested in taking the time" is a bit passive-aggresive sounding to me. Maybe just "want" ?

bradcray replied:
I'm not sure whether you mean "want to take the time..." or "want to understand...", but have bet on the latter for the time being.

e-kayrakli replied:
Yep, that was the right bet.

mppf commented on doc/rst/usingchapel/QUICKSTART.rst:
```diff
@@ -3,161 +3,177 @@
-      make check
+  - set ``CHPL_LLVM=bundled`` to have Chapel build and use the bundled
+    version of LLVM (note that building the bundled version of LLVM
+    can take a *long* time)
```
I'm requesting that we take away the emphasis on long here.

I don't think that the LLVM build takes a *long* time but it does take a long time.

bradcray replied:
How long do you find it takes?  How long is _long_?  I'm not sure I've had a build complete in less than an hour, possibly multiple hours?

mppf replied:
Right, it usually takes me 15 minutes, which is long but not *long*. I think a day would be *long* and while I respect everybody's choice to use or not use parallel builds I think *long* here is pessimistic (at least if we expect the reader might use a parallel build).

bradcray replied:
OK, I definitely agree with removing the emphasis for a 15-minute build.  But... what's your secret?  Even with a parallel make, I don't believe I've gotten under an hour yet (admittedly, it was on chapcs rather than my Mac, which I admittedly became even more gun-shy about after #11404).

mppf replied:
The filesystem speed really matters for such things & I have a decent SSD

bradcray replied:
Go on... So do you use `make -j` or `make -j <n>` and if so, with what value of n?  And are we talking about on a Mac, or another system?

mppf replied:
It's my home PC (running Ubuntu). I tend to use `make -j6` because with more parallelism I run out of memory. (I have 32GB). The system has 8 cores / 16 threads. 15 minutes is an approximate amount, not a measured number.

e-kayrakli commented on doc/rst/usingchapel/QUICKSTART.rst:
```diff
@@ -3,161 +3,177 @@
-   configuration. The *quickstart* script used below is designed to help you
-   get started with Chapel. After that, you may want to switch to a more
-   full-featured configuration. See using-a-more-full-featured-chapel_ below.
+1) If you don't already have the Chapel 1.25 source release, see
```
Should we drop the version name here to make the document more future-proof? Similarly for the filenames below

bradcray replied:
This is admittedly a maintenance issue, but we haven't had a problem with it falling out-of-date, and should have even less of a problem as Max automates things like this.  I think the obvious benefit of naming the version number is that, for some who read this and seem very UNIX-/tech-unsavvy, putting something like `cd chapel-[your-version-number]` would cause them to type that in and then wonder why it doesn't work.  So we've traditionally decided to take the maintenance hit here, and I'd like to advocate that we continue to do so.

e-kayrakli commented on doc/rst/usingchapel/QUICKSTART.rst:
```diff
@@ -3,161 +3,177 @@
    c. Set up your environment for Chapel's Quickstart mode.
-      If you are using a shell other than bash,
-      see quickstart-with-other-shells_ below.
+      If you are using a shell other than ``bash``,
```
FWIW, zsh also can use the same script (as also noted under `quickstart-with-other-shells`)

bradcray replied:
Good point.  So if I'm interpreting you correctly, you're suggesting this should say "other than `bash` or `zsh`"?

e-kayrakli replied:
Yep.

e-kayrakli commented on doc/rst/usingchapel/QUICKSTART.rst:
```diff
@@ -3,161 +3,177 @@
 5) Experiment with Chapel in this Quickstart mode to your heart's
-   content.  Once you are comfortable with Chapel and interested in
+   content.  If you'd like to use this build of Chapel in a different
+   shell / terminal session, see :ref:`using-chapel-in-another-shell`
```
no space around `/` ?

e-kayrakli commented on doc/rst/usingchapel/QUICKSTART.rst:
```diff
@@ -3,161 +3,177 @@
-https://chapel-lang.org/performance.html for performance tips.
+If you plan to do performance studies of Chapel programs, be sure to
+use the full-featured version above and see
+https://chapel-lang.org/perf-tips.html for valuable tips.
```
Based on how `--fast` can be overlooked, we could consider mentioning it here, and referring to perf-tips for further notes (which may not be much today, but still)

bradcray replied:
OK, so maybe:

> If you plan to do performance studies of Chapel programs, be sure to
> use the full-featured version above, to compile with `--fast` once
> your program is correct, and to see
> https://chapel-lang.org/perf-tips.html for other valuable tips.

bradcray replied:
On second thought, I dropped `valuable` now that `--fast` is here (the clearly most valuable).

e-kayrakli commented on doc/rst/usingchapel/building.rst:
```diff
@@ -62,132 +56,156 @@ display its help message using:
+  your path.
+
+* by installing Chapel to a specific location—perhaps one that is
+  already in your path—using the instructions in the next section.
```
Technically not "perhaps", right? It has to be in their PATH to be able avoid setting anything with every new shell session after an install. I slightly prefer if you dropped that em-dash aside or make it a stronger claim e.g. " a specific location that's in your $PATH"

bradcray replied:
Well, you could always create a new location ("I've never installed code in `~/bin/` before, but now am motivated to!") and then add it to your path (everybody has to take that step someday), but I take your point and agree it's nicer to keep it simpler.

e-kayrakli reviewed (approved):
Thanks for the improvements. I don't have any major concern, just some comments inline.

mppf commented on doc/rst/usingchapel/building.rst:
```diff
@@ -62,132 +56,156 @@ display its help message using:
-The Chapel sources are structured so that a GNU-compatible make
+The Chapel sources are structured so that a GNU-compatible ``make``
 utility can be used in any source directory to build the sources
 contained in that directory and its subdirectories.  All of these
```
It's not the case that every Makefile in the project supports all of these projects. In particular, as far as I know, `make check` only works on the toplevel Makefile.

Not sure what the right way to fix that is.

bradcray replied:
I'm going to say "not it!" on this one since it's a pre-existing condition without a clear fix and not part of what I set out to tackle in this edit.  If we wanted to pin it on someone, we could turn to @e-kayrakli as reviewer for #17442, but I think it's close enough to the truth that given the very small number of people who probably read this section that it's probably not a big problem.  Certainly not something that gives me any pause.

mppf commented on doc/rst/usingchapel/building.rst:
```diff
@@ -62,132 +56,156 @@ display its help message using:
 The Chapel makefiles have a few options that enable or disable optimization,
-debugging support, profiling, and backend C compiler warnings. The variables
+debugging support, profiling, and back-end C compiler warnings. The variables
 are described below. Set the value to 1 to enable the feature.
```
I think this would benefit from an example like 

```
Set the value to 1 to enable the feature (e.g. `make OPTIMIZE=1 WARNINGS=1`)
```

mppf reviewed (approved):
Thanks for your effort at improving these! I read through the current versions (without regard to what was new and what was old) and have some feedback in line-comments here.

bradcray commented on doc/rst/usingchapel/QUICKSTART.rst:
```diff
@@ -3,161 +3,177 @@
 
-      # re-build Chapel
-      make
+  - ensure that you have a version of LLVM 11 installed on your system
```
@e-kayrakli : This is the "embedding a version number" that I feel more nervous will fall out of date.  But similar to the previous argument, I fear that not saying it will cause people to say "Hey, I have LLVM, but it didn't work, what gives?"

bradcray commented:
@mppf: 

> Perhaps you were imagining that make install is a way to "set up your environment"?

I definitely was, but I like your approach better.  And I can raise you one as well and mention `./configure` and `make install` here as well which I think would actually have helped with the OP of #18150 (he could've searched on the strings and found them for example.  So I've currently got:

```
Note that in both the Quickstart and preferred modes above, any
environment settings made by ``setchplenv.bash`` will not persist
beyond your current shell/terminal session.  One easy way to use
Chapel from a different shell or terminal is to re-``source`` the
``setchplenv.bash`` script that you used when building Chapel.
However since this can quickly become annoying, other strategies are
available including a ``./configure`` + ``make install`` option.  See
:ref:`using-chapel-in-other-sessions` for details.
```

if that works for you?

mppf commented:
@bradcray - yeah that works for me.
